src/NUnitFramework/framework/Attributes/RangeAttribute.cs

```diff
@@ -192,12 +202,76 @@ public RangeAttribute(float from, float to, float step)
             float aStep = Math.Abs(step);
             float tol = aStep / 1000;
             int count = (int)(Math.Abs(to - from) / aStep + tol + 1);
-            this.data = new object[count];
+            _data = new object[count];
             int index = 0;
             for (float val = from; index < count; val += step)
-                this.data[index++] = val;
+                _data[index++] = val;
         }
 
         #endregion
+
+        /// <summary>
+        /// Get the range of values to be used as arguments
+        /// </summary>
+        public IEnumerable GetData(IParameterInfo parameter)
+        {
+            Type tartgetType = parameter.ParameterType;
+
+            if (tartgetType.GetTypeInfo().IsEnum && _data.Length == 0)
+            {
+                return Enum.GetValues(tartgetType);
+            }
+
+            if (tartgetType == typeof(bool) && _data.Length == 0)
+            {
+                return new object[] { true, false };
+            }
+
+            return GetData(tartgetType);
+        }
+
+        private IEnumerable GetData(Type targetType)
+        {
+            for (int i = 0; i < _data.Length; i++)
+            {
+                object arg = _data[i];
+
+                if (arg == null)
+                {
+                    continue;
+                }
+
+                if (targetType.GetTypeInfo().IsAssignableFrom(arg.GetType().GetTypeInfo()))
+                {
+                    continue;
+                }
+
+#if !NETSTANDARD1_3 && !NETSTANDARD1_6
+                if (arg is DBNull)
+                {
+                    _data[i] = null;
+                    continue;
+                }
+#endif
+
+                var convert = false;
+
+                if (targetType == typeof(short) || targetType == typeof(byte) || targetType == typeof(sbyte))
+                {
+                    convert = arg is int;
+                }
+                else if (targetType == typeof(decimal))
+                {
+                    convert = arg is double || arg is int;
+                }
+
```

Comment:
Can you clarify why you removed TimeSpan and DateTime here? It's possible to use range attribute with both of those values...isn't it?

Reply:
`RangeAttribute` has a fixed set of constructors for specific numeric types. Supporting TimeSpan and DateTime would be an enhancement requiring a bit of new logic. Nobody has requested it and it would be better to do separately anyway, I think.

Reply:
Ah - never mind, you're right - turns out you can't use RangeAttribute with TimeSpan or DateTime!

Reply:
We cross-posted @CharliePoole, but yes - sorry, I'd mistakenly remembered this already being a feature!